### PR TITLE
[release/v7.3.0-rc.1] Replace win7 runtime with win8 and remove APISets (#18304)

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Backport change #18304 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by xtqqczze** [comment]
@adityapatwardhan Windows Server 2012 is still supported under Extended Security Updates until [October 13, 2026](https://learn.microsoft.com/en-us/windows-server/get-started/extended-security-updates-overview#what-are-extended-security-updates), and will be supported in .NET throughout .NET 7 and .NET 8, according to dotnet/core#7556.

Windows Server 2012 uses `win8-*` RIDs.

File: .vsts-ci/windows/templates/windows-packaging.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ parameters:
   - name: jobName
     default: 'win_packaging'
   - name: runtimePrefix
-    default: 'win7'
+    default: 'win81'
```
Comment (xtqqczze) [inline .vsts-ci/windows/templates/windows-packaging.yml:7]:
```suggestion
    default: 'win8'
```

To support Windows Server 2012.

**Review by xtqqczze — CHANGES_REQUESTED** [review]

File: .vsts-ci/windows/templates/windows-packaging.yml (hunk 1 above)
Comment (adityapatwardhan) [reply .vsts-ci/windows/templates/windows-packaging.yml:7]: See the PR https://github.com/PowerShell/PowerShell/pull/18322. We reverted win7 -> win81 change.  win7 will still not be supported.

Comment (xtqqczze) [reply .vsts-ci/windows/templates/windows-packaging.yml:7]: But will we support Windows Server 2012? see: https://github.com/PowerShell/PowerShell/pull/18354#issuecomment-1287899773.

Comment (SteveL-MSFT) [reply .vsts-ci/windows/templates/windows-packaging.yml:7]: @adityapatwardhan I think what @xtqqczze is saying is that to support Server 2012 (which is still under Extended support), we need `win8`, not `win81`

**Comment by daxian-dbw** [comment]
Are you backporting #18304 just to prepare for backporting the PR that reverted it?

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
#18322 is a partial revert.

**Pull request merged**